### DISCUSSION TITLE
How to control the temperature of the LLM.The goal is to ensure consistent answers in each conversation

### DISCUSSION THREAD

**Discussion by Frog863** [crewAIInc/crewAI#1197] [Q&A]

**Comment by zinyando** [comment]
You set the temperature when you create the LLM object eg

```python
from langchain_openai import ChatOpenAI

llm = ChatOpenAI(
    model="gpt-4o",
    temperature=0,
    max_tokens=None,
    timeout=None,
    max_retries=2,
    # api_key="...",
)
```

**Comment by Frog863** [comment]
Thank you!

**Comment by gadgethome** [comment]
Also, have a look at the new training feature. 
https://docs.crewai.com/core-concepts/Training-Crew/?h=train

**Comment by Frog863** [comment]
I set LLM temperature=0, but the output is still different.Does Crewai have models built in? So how to ensure that crewai's model input and output are unchanged. Is it OK to set a fixed random seed？

**Reply by zinyando** [reply]
> Does Crewai have models built in?

CrewAI doesn't have built-in models, you bring your own model.

> So how to ensure that crewai's model input and output are unchanged?

LLMs are not completely deterministic. The same input might yield different results, setting the temperature makes it better by making the results as close together as possible.

> Is it OK to set a fixed random seed？

Unlike Autogen, CrewAI doesn't currently have the cache/seed concept. You can use the training feature mentioned above to train your crew to produce results that you like.

**Reply by Frog863** [reply]
Ok,thank you!

**Reply by theCyberTech** [reply]
Also be sure to give examples of what you expect in the task expected_output attribute, that will help some
